lib/full-site-editing/block-templates.php

```diff
@@ -173,6 +173,17 @@ function _gutenberg_build_template_result_from_file( $template_file, $template_t
 		$template->title       = $default_template_types[ $template_file['slug'] ]['title'];
 	}
 
+	if ( 'wp_template_part' === $template_type && is_readable( locate_template( 'experimental-theme.json' ) ) ) {
+		$theme_json = file_get_contents( locate_template( 'experimental-theme.json' ) );
+		$data       = json_decode(
+			$theme_json,
+			true
+		);
+		if ( isset( $data['template-parts'][ $template_file['slug'] ]['type'] ) ) {
+			$template->template_part_type = $data['template-parts'][ $template_file['slug'] ]['type'];
```

Comment:
It seems type might be redundunt with the "type" above. I wonder if we could find a better for this property. "scope" maybe? cc @mtias @mcsf

Reply:
I'm leaning towards "section"

Reply:
After stepping away I starting thinking `semantic_type` would fit better for both "type" and "template_part_type".  But if we end up allowing types such as "left sidebar" or things that don't really line up with semantic HTML elements 1-1, something like "section" seems fitting.